### PR TITLE
fix: correct mulligan state for spectated matches

### PR DESCRIPTION
**Pull request by rikumiyao**
Currently it appears that the GameEntity BEGIN_MULLIGAN event is not
being emitted in the Hearthstone log for spectated matches. We should
instead rely on the MULLIGAN_STATE events, which should be outputted
twice for each player per game.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,32 +23,32 @@
...
     "npm-run-all": "^4.1.5",
-    "nyc": "^14.0.0",
+    "nyc": "^14.1.1",
     "semantic-release": "^15.13.14",
     "source-map-support": "^0.5.12",
     "trash-cli": "^1.4.0",
-    "ts-node": "^8.0.3",
-    "typescript": "^3.4.3"
+    "ts-node": "^8.1.0",
+    "typescript": "^3.4.5"
```
Comment (Lange) [inline package.json]: Can these changes to `package.json` and `package-lock.json` be done in a separate PR?

File: src/line-parsers/game-tag-change.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export class GameTagChangeLineParser extends AbstractLineParser {
 			gameState.turnStartTime = new Date();
 		}
 
-		if (data.entity === 'GameEntity' && data.tag === 'STEP' && data.value === 'BEGIN_MULLIGAN') {
+		if (data.tag === 'MULLIGAN_STATE' && data.value === 'DEALING') {
```
Comment (Lange) [inline src/line-parsers/game-tag-change.ts]: Awesome! Glad the fix is so simple.

File: package.json (hunk 1 above)
Comment (rikumiyao) [reply package.json]: It appears that the master branch is already failing tests giving the error
```npm ERR! audit Your configured registry (https://registry.npmjs.org/) does not support audit requests.```
which I fixed by deleteing node_modules and reinstalling packages, but I guess we should merge the package.json changes first then?

Comment (Lange) [reply package.json]: Oh, that's a random failure, it's a super roundabout way of npm saying "hey our server gave an error response, try again in a bit". You can most likely discard all these `package.json` and `package-lock.json` changes.

File: src/line-parsers/game-tag-change.ts (hunk 2 above)
Comment (rikumiyao) [reply src/line-parsers/game-tag-change.ts]: Yes, the MAIN_READY event is emitted regardless of whether we are spectating are not.
